MathlibTest/DeprecatedSyntaxLinter.lean

```diff
@@ -83,6 +83,65 @@ warning: declaration uses 'sorry'
 #guard_msgs in
 example : False := by admit
 
+set_option linter.style.nativeDecide true
+
+/--
+warning: Using `native_decide` is not allowed in mathlib:
+because it trusts the entire Lean compiler (not just the Lean kernel),
+it could quite possibly be used to prove false.
+note: this linter can be disabled with `set_option linter.style.nativeDecide false`
+-/
+#guard_msgs in
+example : 1 + 1 = 2 := by native_decide
+
+/--
+warning: Using `decide +native` is not allowed in mathlib:
+because it trusts the entire Lean compiler (not just the Lean kernel),
+it could quite possibly be used to prove false.
+note: this linter can be disabled with `set_option linter.style.nativeDecide false`
+-/
+#guard_msgs in
+example : 1 + 1 = 2 := by decide +native
+#guard_msgs in
+example : 1 + 1 = 2 := by decide -native
+
+/--
+warning: Using `decide +native` is not allowed in mathlib:
+because it trusts the entire Lean compiler (not just the Lean kernel),
+it could quite possibly be used to prove false.
+note: this linter can be disabled with `set_option linter.style.nativeDecide false`
+-/
+#guard_msgs in
+theorem foo : 1 + 1 = 2 := by decide -native +native
+#guard_msgs in
+example : 1 + 1 = 2 := by decide +native -native
+
+/--
+warning: Using `decide +native` is not allowed in mathlib:
+because it trusts the entire Lean compiler (not just the Lean kernel),
+it could quite possibly be used to prove false.
+note: this linter can be disabled with `set_option linter.style.nativeDecide false`
+-/
+#guard_msgs in
+example : 1 + 1 = 2 := by decide +native -native +native +native
+#guard_msgs in
+example : 1 + 1 = 2 := by decide +native -native +kernel +native -kernel +native -native
+
+/--
+warning: Using `decide +native` is not allowed in mathlib:
+because it trusts the entire Lean compiler (not just the Lean kernel),
+it could quite possibly be used to prove false.
+note: this linter can be disabled with `set_option linter.style.nativeDecide false`
+-/
+#guard_msgs in
+example : 1 + 1 = 2 := by decide (config := { native := true })
+example : 1 + 1 = 2 := by decide (config := { native := false })
+-- Not handled yet: since mathlib hardly uses the old config syntax and this linter is purely
+-- for user information (and not hard guarantees), we deem this acceptable.
+example : 1 + 1 = 2 := by decide (config := { native := false, kernel := false })
```

Comment:
I think this should be accepted by lean4checker, right? The comment seems to be describing `config := { native := true, kernel := false }`, which would be rejected by lean4checker but accepted by the new linter